lib/Translation/ExportVerilog/ExportVerilog.cpp

```diff
@@ -2658,6 +2669,21 @@ void ModuleEmitter::prepareHWModule(Block &block) {
       op.erase();
       continue;
     }
+
+    // Name legalization should have happened in a different pass for these sv
+    // elements and we don't want to change their name through re-legalization
+    // (e.g. letting a temporary take the name of an unvisited wire). Adding
+    // them now ensures any temporary generated will not use one of the names
+    // previously declared.
+    if (auto instance = dyn_cast<InstanceOp>(op))
+      if (!addLegalName(ValueOrOp(&op), instance.instanceName()))
+        emitOpError(instance, "Instance name changed durring emission.");
+    if (auto wire = dyn_cast<WireOp>(op))
+      if (!addLegalName(ValueOrOp(op.getResult(0)), wire.name()))
+        emitOpError(wire, "Wire name changed durring emission");
```

Comment:
Did you consider passing in the op and error message as a const char * to "addLegalName"?  This would allow it to emit the error and reduce boilerplate.